modules/lang-painless/src/test/java/org/elasticsearch/painless/RegexTests.java

```diff
@@ -232,6 +232,30 @@ public void testReplaceFirstQuoteReplacement() {
                 exec("'the quick brown fox'.replaceFirst(/[aeiou]/, m -> '$' + m.group().toUpperCase(Locale.ROOT))"));
     }
 
+    public void testStoreInMap()  {
```

Comment:
I think it might be useful to have similar tests in the constant folding tests that verify that we fold constants in these positions.

Reply:
We could use the bytecode assertion for that kind of test.